static/js/settings_notifications.js

```diff
@@ -121,7 +137,11 @@ exports.update_page = function () {
             // If push notifications are disabled at the realm level,
             // we should just leave the checkbox always off.
             return;
+        } else if (setting === 'desktop_icon_count_display') {
+            update_desktop_icon_count_display();
+            return;
```

Comment:
This code is a little goofy in that we update the state of all settings whenever any one is changed, but I supposed that's pre-existing.